ospfd/ospf_vty.c

```diff
@@ -6281,23 +6716,43 @@ static void show_ip_ospf_database_maxage(struct vty *vty, struct ospf *ospf)
 static int show_ip_ospf_database_common(struct vty *vty, struct ospf *ospf,
 					int arg_base, int argc,
 					struct cmd_token **argv,
-					uint8_t use_vrf)
+					uint8_t use_vrf, json_object *json,
+					bool uj)
 {
 	int idx_type = 4;
 	int type, ret;
 	struct in_addr id, adv_router;
+	json_object *json_vrf = NULL;
 
-	if (ospf->instance)
-		vty_out(vty, "\nOSPF Instance: %d\n", ospf->instance);
+	if (uj) {
+		if (use_vrf)
+			json_vrf = json_object_new_object();
+		else
+			json_vrf = json;
+	}
 
-	ospf_show_vrf_name(ospf, vty, NULL, use_vrf);
+	if (ospf->instance) {
+		if (uj)
+			json_object_int_add(json, "ospfInstance",
+					    ospf->instance);
+		else
+			vty_out(vty, "\nOSPF Instance: %d\n\n", ospf->instance);
+	}
 
-	vty_out(vty, "\n       OSPF Router with ID (%s)\n\n",
-		inet_ntoa(ospf->router_id));
+	ospf_show_vrf_name(ospf, vty, json, use_vrf);
+
+	/* Show Router ID. */
+	if (uj) {
+		json_object_string_add(json_vrf, "routerId",
+				       inet_ntoa(ospf->router_id));
+	} else {
+		vty_out(vty, "\n       OSPF Router with ID (%s)\n\n",
+			inet_ntoa(ospf->router_id));
+	}
 
 	/* Show all LSA. */
-	if (argc == arg_base + 4) {
-		show_ip_ospf_database_summary(vty, ospf, 0);
+	if ((argc == arg_base + 4) || (uj && (argc == arg_base + 5))) {
+		show_ip_ospf_database_summary(vty, ospf, 0, json_vrf);
```

Comment:
Since we are returning from this function here. We need to add the following code here.
		if (use_vrf) {
			if (ospf->vrf_id == VRF_DEFAULT)
				json_object_object_add(json, "default",
						       json_vrf);
			else
				json_object_object_add(json, ospf->name,
						       json_vrf);
		}

Reply:
Done